src/main/controllers/AttachmentController.ts

```diff
@@ -22,24 +22,43 @@ export default class AttachmentController {
     let decisionDocId;
     let requestDocId;
     let contactTribunalDocId;
+    const adminDecision = userCase.selectedGenericTseApplication?.value.adminDecision;
 
     if (req.session.documentDownloadPage === PageUrls.RESPONDENT_APPLICATION_DETAILS) {
       respondentAppDocId = getDocId(userCase.selectedGenericTseApplication?.value.documentUpload.document_url);
+
+      claimantResponseDocId = undefined;
       const claimantResponseDoc = getClaimantResponseDocDownload(userCase.selectedGenericTseApplication);
-      claimantResponseDocId = getDocId(claimantResponseDoc.document_url);
-      decisionDocId = userCase.selectedGenericTseApplication?.value.adminDecision
-        .map(it => getDocId(it.value.responseRequiredDoc.document_url))
-        .find(id => id === docId);
+      if (claimantResponseDoc !== undefined) {
```

Comment:
Could you please refactor these sections into methods like getClaimantResponseDocId or getDecisionDocId. It's nicer to move the logic out of the controller to helpers. That way we can just read through the methods called in the controller and be able to figure out what's happening in each step